lib/collection/src/shards/replica_set/partial_snapshot_meta.rs

```diff
@@ -27,12 +27,15 @@ use crate::operations::types::{CollectionError, CollectionResult};
 ///
 #[derive(Debug, Default)]
 pub struct PartialSnapshotMeta {
-    /// Tracks ongoing operations of creation of partial snapshots.
-    /// There might be multiple parallel requests to create partial snapshots,
-    /// so we track them with a counter.
+    /// Tracks ongoing *create* partial snapshot requests. There might be multiple parallel
+    /// create partial snapshot requests, so we track them with a counter.
     ongoing_create_snapshot_requests_tracker: RequestTracker,
-    recovery_lock: Arc<tokio::sync::Mutex<()>>,
-    search_lock: Arc<tokio::sync::RwLock<()>>,
+
+    /// Limits parallel *recover* partial snapshot requests. We are using `RwLock`, so that multiple
+    /// read requests can check if recovery is in progress (by doing `try_read`) without blocking
+    /// each other.
+    recovery_lock: Arc<tokio::sync::RwLock<()>>,
```

Comment:
We now (sometimes) check if `recovery_lock` is locked during `execute_local_read_operation`, and so I've swapped `Mutex` for `RwLock`, so that concurrent read operations can do `try_read` without blocking each other.